src/ParseFile.js

```diff
@@ -348,33 +355,43 @@ const DefaultController = {
     return CoreManager.getRESTController().request('POST', path, data, options);
   },
 
-  download: function(uri) {
+  download: function(uri, options) {
     if (XHR) {
-      return this.downloadAjax(uri);
+      return this.downloadAjax(uri, options);
     } else if (process.env.PARSE_BUILD === 'node') {
       return new Promise((resolve, reject) => {
+        let aborted = false;
         const client = uri.indexOf('https') === 0
           ? require('https')
           : require('http');
-        client.get(uri, (resp) => {
+        const req = client.get(uri, (resp) => {
           resp.setEncoding('base64');
           let base64 = '';
           resp.on('data', (data) => base64 += data);
           resp.on('end', () => {
+            if (aborted) {
+              return resolve({});
```

Comment:
on first reading, I got confused by the return here.  I think it would be more clear if you just `resolve({})` without the `return`.

also, instead of having a variable `abort`, couldn't you just `resolve` in the `abort` event handler on [381.](https://github.com/parse-community/Parse-SDK-JS/pull/951/files#diff-67ccbb663bacb4c7556216ddb313067cR381)?

Reply:
I believe when you abort, the abort handler and end event both gets called. If I call resolve in abort would end still get called?

Reply:
I believe it would, but easy enough to test?